dll/cpl/sysdm/environment.c

```diff
@@ -45,97 +187,779 @@ GetSelectedListViewItem(HWND hwndListView)
     return -1;
 }
 
+static LRESULT CALLBACK
+ListViewSubclassProc(HWND hListBox,
+                    UINT uMsg,
+                    WPARAM wParam,
+                    LPARAM lParam,
+                    UINT_PTR uIdSubclass,
+                    DWORD_PTR dwRefData)
+{
+    switch (uMsg)
+    {
+        case WM_DESTROY:
+        {
+            RemoveWindowSubclass(hListBox, ListViewSubclassProc, uIdSubclass);
+            break;
+        }
+
+        /* Whenever the control is resized make sure it doesn't spawn the horizontal scrollbar */
+        case WM_SIZE:
+        {
+            ShowScrollBar(hListBox, SB_HORZ, FALSE);
+            break;
+        }
+    }
+
+    return DefSubclassProc(hListBox, uMsg, wParam, lParam);
+}
+
+static VOID
+AddEmptyItem(HWND hwndListView,
+             DWORD dwSelectedValueIndex)
+{
+    LV_ITEM lvi;
+
+    ZeroMemory(&lvi, sizeof(lvi));
+    lvi.mask = LVIF_TEXT | LVIF_STATE;
+    lvi.cchTextMax = MAX_PATH;
+    lvi.pszText = _T("");
+    lvi.iItem = dwSelectedValueIndex;
+    lvi.iSubItem = 0;
+    ListView_InsertItem(hwndListView, &lvi);    
+}
+
+static VOID
+AddValuesToList(HWND hwndDlg,
+                   PDIALOG_DATA DlgData)
+{
+    LV_COLUMN column;
+    LV_ITEM lvi;
+    RECT rItem;
+
+    DWORD dwValueLength;
+    DWORD i;
+    HWND hwndListView;
+    LPTSTR lpTemp;
+    LPTSTR lpToken;
+
+    ZeroMemory(&column, sizeof(column));
+    ZeroMemory(&lvi, sizeof(lvi));
+
+    hwndListView = GetDlgItem(hwndDlg, IDC_LIST_VARIABLE_VALUE);
+
+    GetClientRect(hwndListView, &rItem);
+
+    column.mask = LVCF_WIDTH;
+    column.cx = rItem.right;
+    ListView_InsertColumn(hwndListView, 0, &column);
+    ShowScrollBar(hwndListView, SB_HORZ, FALSE);
+
+    lvi.mask = LVIF_TEXT | LVIF_STATE;
+    lvi.cchTextMax = MAX_PATH;
+    lvi.iSubItem = 0;
+
+    dwValueLength = _tcslen(DlgData->VarData->lpRawValue) + 1;
+    lpTemp = GlobalAlloc(GPTR, dwValueLength * sizeof(TCHAR));
+
+    StringCchCopy(lpTemp, dwValueLength, DlgData->VarData->lpRawValue);
+    lpToken = _tcstok(lpTemp, _T(";"));
+
+    i = 0;
+
+    while (lpToken != NULL)
```

Comment:
Random suggestion if you like it:
```suggestion

    for (lpToken = _tcstok(lpTemp, _T(";")), i = 0;
         lpToken != NULL;
         lpToken = _tcstok(NULL, _T(";")), i++)
```

Reply:
@HBelusca Interesting, never seen someone using two variables at the same time for "for loop".

Reply:
Seems like it was only gather *one* value hence I decided to not change it.

Reply:
Otherwise this for-loop is equivalent to the existing while loop one.